### ISSUE TITLE
Incorrect `repoSize` when using symbolic links

### ISSUE THREAD

**Issue by bitspill** [ipfs/go-ipfs#4056]
#### Version information:
```
go-ipfs version: 0.4.10-
Repo version: 5
System version: amd64/linux
Golang version: go1.8.3
```

#### Type:
`Bug`

#### Severity:
`Low`

#### Description:

If the .ipfs directory is a symbolic link the repoSize as returned by `ipfs repo stat` or `ipfs stats repo` will always return 9 bytes regardless of actual data stored

With symlink
```
~$ ls -al
lrwxrwxrwx   1 ipfs ipfs        9 Jun 22  2016 .ipfs -> ipfs_data
drwxrwxr-x   5 ipfs ipfs     4096 Jul 12 01:25 ipfs_data

~$ ipfs repo stat
NumObjects: 204114
RepoSize:   9
StorageMax: 800000000000
RepoPath:   /home/ipfs/.ipfs
Version:    fs-repo@5
```

Without symlink
```
~$ ipfs repo stat
NumObjects: 204114
RepoSize:   49450734482
StorageMax: 800000000000
RepoPath:   /home/ipfs/.ipfs
Version:    fs-repo@5
```

**Comment by bitspill** [comment]
Did some looking and found that filepath.Walk does not follow symbolic links

[1] https://github.com/ipfs/go-ipfs/blob/master/repo/fsrepo/fsrepo.go#L591
[2] https://golang.org/pkg/path/filepath/#Walk

**Comment by leerspace** [comment]
This was fixed in #4305
